### ISSUE TITLE
Trouble training YOLOE with visual prompts on custom segmentation data

### ISSUE THREAD

**Issue by misabellerv** [ultralytics/ultralytics#20822]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello!

My team and I recently decided to work with YOLOE for visual prompt input and automatic segmentation based on our custom dataset. We are working with wood boards and have previously trained YOLOv8 models for detection/segmentation of classes such as full wood, top face, and side face. The new challenge is to use YOLOE for auto-annotation of segmentation masks from visual prompt inputs.

However, the documentation is not very clear on how to properly train YOLOE with custom data and perform inference using visual prompts (e.g., bounding boxes) to obtain the corresponding segmentation output. The official Ultralytics website provides different training methods, and some GitHub issues mention different trainers for the model. Here is the training code we are using:

```python 
from ultralytics import YOLOE
from ultralytics.models.yolo.yoloe import YOLOESegVPTrainer

data = dict(
    train=dict(yolo_data=["/home/isabelle/YOLOe/data.yaml"]),
    val=dict(yolo_data=["/home/isabelle/YOLOe/data.yaml"]),
)

model = YOLOE("yoloe-11l-seg.pt")

model.train(
    data=data,
    batch=64,
    epochs=100,
    close_mosaic=2,
    optimizer="AdamW",
    lr0=1e-4,
    warmup_bias_lr=0.0,
    weight_decay=0.025,
    momentum=0.9,
    workers=4,
    trainer=YOLOESegVPTrainer,  
    device="0,1,2,3",
    mask_ratio=1,
    overlap_mask=True
)
```
And for predictions:

```python 
model = YOLOE(best_model_path)

results = model.predict(
    image_path,
    visual_prompts=visual_prompts,
    predictor=YOLOEVPSegPredictor,
    retina_masks=True,
)
```
Our data.yaml includes polygon annotations (masks) for segmentation:

```yaml
path: /home/isabelle/YOLOe/data
train: /home/isabelle/YOLOe/data/images/train
val: /home/isabelle/YOLOe/data/images/validation
test: /home/isabelle/YOLOe/data/images/test

nc: 3

names:
  0: Top_Face
  1: Full_Wood
  2: Side_Face
```
From what we understand, we're not using frozen modules in the model and we're training everything—from the backbone to the visual prompt module—to specialize the model on our dataset. Is this the correct approach? Is this the recommended way to train YOLOE when the final goal is to provide a visual prompt and get segmentation predictions? 

Another question on model performance (and one of the reasons we are curious to know if there's another - and more suitable - training approach): Using parameters similar to those used in our successful YOLOv8 training, the performance of YOLOE has been poor. We are seeing a high number of false negatives, and the loss struggles to converge—both for bounding boxes and segmentation. The results are far from production-ready, even that for raw YOLOv8 the results are excellent. Could this behavior be expected? Or is there something fundamentally wrong with how we are training/inferencing, based on the code we shared?

Any guidance or clarification would be greatly appreciated. Thank you in advance!

**Comment by Y-T-G** [comment]
> From what we understand, we're not using frozen modules in the model and we're training everything—from the backbone to the visual prompt module—to specialize the model on our dataset. Is this the correct approach? Is this the recommended way to train YOLOE when the final goal is to provide a visual prompt and get segmentation predictions?

You should be freezing the modules as shown in the guide.
https://docs.ultralytics.com/models/yoloe/#__tabbed_4_2

You don't train the whole model during visual prompt based training.

**Comment by Y-T-G** [comment]
Also have you tested the YOLOE visual prompts without training on your dataset? The model was trained on large and diverse datasets. Training it on smaller and less diverse dataset would probably yield worse results.

You should just fine-tune the model in that case:
https://docs.ultralytics.com/models/yoloe/#__tabbed_1_1
